src/main/java/com/joejoe2/chat/exception/ChannelDoesNotExist.java

```diff
@@ -1,7 +1,11 @@
 package com.joejoe2.chat.exception;
 
 public class ChannelDoesNotExist extends Exception {
-  public ChannelDoesNotExist(String channelId) {
-    super("channel %s does not exist !".formatted(channelId));
+  public ChannelDoesNotExist(String message) {
+    super(message);
+  }
+
+  public static ChannelDoesNotExist ofId(String channelId) {
```

Comment:
By introducing this things became more complex as these could be. One of the most important rules of Software Engineering is to keep It simple.

We can simplify this by just passing a corresponding exception message while throwing an exception. That's totally fine.